crates/pera-core/src/authority.rs

```diff
@@ -1522,6 +1533,28 @@ impl AuthorityState {
         Ok(())
     }
 
+    async fn handle_mpc_events(
+        &self,
+        inner_temporary_store: &InnerTemporaryStore,
+        effects: &TransactionEffects,
+        epoch_store: &Arc<AuthorityPerEpochStore>,
+    ) -> anyhow::Result<()> {
+        if !self.is_validator(epoch_store) {
+            return Ok(());
+        }
+        let status = match &effects {
+            TransactionEffects::V1(effects) => effects.status(),
+            TransactionEffects::V2(effects) => effects.status(),
+        };
+        if status.is_err() {
+            return Ok(());
+        }
+        self.mpc_state
+            .lock()
+            .handle_mpc_events(&inner_temporary_store.events.data)?;
+        Ok(())
```

Comment:
IIUC, you can simplify this to

```suggestion
        self.mpc_state
            .lock()
            .handle_mpc_events(&inner_temporary_store.events.data)
```